### PR TITLE
fix: Correctly disconnect user

### PR DESCRIPTION
**Pull request by PhilippeWeidmann**

### CODE REVIEW

File: kDrive/AppDelegate.swift [hunk 1]
```diff
@@ -82,6 +83,8 @@ final class AppDelegate: UIResponder, UIApplicationDelegate, AccountManagerDeleg
         let currentState = RootViewControllerState.getCurrentState()
         prepareRootViewController(currentState: currentState)
 
+        (accountManager as? AccountManager)?.delegate = self
```
Comment (PhilippeWeidmann) [inline kDrive/AppDelegate.swift:86]: If I don't cast, i get a crash in the DI. Do you know why @adrien-coye ?

Comment (adrien-coye) [reply kDrive/AppDelegate.swift:86]: You cannot declare weak properties in a pure swift protocol (even if inherits from `AnyObject`)
You probably want to stick to a `setDelegate(_ )` method for this case.

You can declare a property weak in a protocol only if @Objc which we probably do not want.

File: kDrive/AppDelegate.swift [hunk 2]
```diff
@@ -285,8 +291,11 @@ final class AppDelegate: UIResponder, UIApplicationDelegate, AccountManagerDeleg
     func refreshCacheData(preload: Bool, isSwitching: Bool) {
         Log.appDelegate("refreshCacheData preload:\(preload) isSwitching:\(preload)")
 
-        @InjectService var accountManager: AccountManageable
-        let currentAccount = accountManager.currentAccount!
+        guard let currentAccount = accountManager.currentAccount else {
+            Log.appDelegate("No account to refresh")
```
Comment (adrien-coye) [inline kDrive/AppDelegate.swift:295]:
```suggestion
            Log.appDelegate("No account to refresh", level: .error)
```

File: kDrive/AppDelegate.swift [hunk 3]
```diff
@@ -458,7 +465,9 @@ final class AppDelegate: UIResponder, UIApplicationDelegate, AccountManagerDeleg
     // MARK: - Account manager delegate
 
     func currentAccountNeedsAuthentication() {
-        setRootViewController(SwitchUserViewController.instantiateInNavigationController())
+        DispatchQueue.main.async { [weak self] in
```
Comment (adrien-coye) [inline kDrive/AppDelegate.swift:468]: Can we use `Task { @MainActor _ `instead ?

**Review by adrien-coye — APPROVED** [review]
Code as is looks good. Two remarks if you agree with me, you can make two small changes.

**Review by adrien-coye — APPROVED** [review]

**Pull request merged**